tensorboard/plugins/profile/input_pipeline_analyzer/input-pipeline-analyzer.html

```diff
@@ -0,0 +1,373 @@
+      this.set('_steptime_ms_maximum', deviceJson.p['steptime_ms_maximum']);
+      this.set('_host_conclusion', hostJson.p['conclusion']);
+      this.customStyle['--summary-color'] =  deviceJson.p['summary_color'];
+      this.updateStyles();
+      this._showDeviceStepChart(deviceJson);
+      this._showDeviceInfeedChart(deviceJson);
+      this._showHostChart(hostJson);
+      this._showHostTable(hostJson);
+    },
+
+    /* Plots a graph of step time (in ms) against step number */
+    _showDeviceStepChart : function(deviceJson) {
+      var deviceStepSeries = [];
+      var hostStepSeries = [];
+      var maxStepTime = 0;
+      deviceJson.rows.forEach(function(row, index) {
+        var step = Number(row.c[0].v);
+        deviceStepSeries.push({"scalar": row.c[1].v,
+                               "step": step,
+                               "low_watermark": 0})
+        hostStepSeries.push({"scalar": row.c[1].v + row.c[2].v,
+                             "step": step,
+                             "low_watermark": row.c[1].v})
+        maxStepTime = Math.max(maxStepTime, row.c[1].v + row.c[2].v)
+      })
+      var chart = document.querySelector('#device_step_chart');
+      if (chart) {
+        chart.setVisibleSeries(['total step time', 'device step time']);
+        chart.setSeriesData('total step time', hostStepSeries);
+        chart.setSeriesData('device step time', deviceStepSeries);
+        chart.defaultYRange = [0, maxStepTime * 1.1]
+        chart.smoothingEnabled = false;
+        chart.tooltipColumns = [
+          { title: 'Name', evaluate: (d) => d.dataset.metadata().name },
+          { title: 'Time(ms)', evaluate: (d) => d.datum.scalar },
+          { title: 'Step', evaluate: (d) => d.datum.step }];
+        chart.fillArea = {
+          higherAccessor: (d) => d.scalar,
+          lowerAccessor: (d) => d.low_watermark,
+        };
+      }
+    },
+
+    /* Plots a graph of infeed dequeue time (in % of the step time) against step number */
+    _showDeviceInfeedChart: function (deviceJson) {
+      var deviceInfeedSeries = [];
+      deviceJson.rows.forEach(function(row, index) {
+        deviceInfeedSeries.push({"step": Number(row.c[0].v),
+                                 "scalar": row.c[4].v,
+                                 "min": row.c[5].v,
+                                 "max": row.c[6].v});
+      })
+      var chart = document.querySelector('#device_infeed_chart');
+      if (chart) {
+        chart.setVisibleSeries(['infeed deque time%']);
+        chart.setSeriesData('infeed deque time%', deviceInfeedSeries);
+        chart.tooltipColumns = [
+          { title: 'Name', evaluate: (d) => d.dataset.metadata().name },
+          { title: 'Step', evaluate: (d) => d.datum.step },
+          { title: 'Average(%)', evaluate: (d) => (d.datum.scalar).toFixed(2) + '%' },
+          { title: 'Min(%)', evaluate: (d) => (d.datum.min).toFixed(2) + '%' },
```

Comment:
nit: Could we introduce a helper function `_formatFloat` and then use that instead where `.toFixed(2)` is currently used throughout this file?

Reply:
it is coincidence now all float are converted with same precision after decimal point, however I don't think it will be the case in the future.